### PR TITLE
dev-feature-info-agent: 3.0.0

### PR DESCRIPTION
**Pull request by mdhillman**
This PR brings in a number of changes to the FeatureInfoAgent (detailed below with content from the CHANGELOG file).

A sample stack (for quick FIA testing, not to be considered a nice example usage) has been produced, read how to run it from the `Agents/FeatureInfoAgent/sample` directory. If possible, these changes should be tested with real use-case data (understood that is difficult to get real time series data, hence the sample stack). I have tested this new version with the sample stack and pre-3.0.0 configurations using the UK Base World Visualisation.

Original plans included creating an FIA tutorial, however I think this is best done after changes to the time series client has been carried out (and when more resource is available).

A docker image for testing has been built and pushed, name and tag is:

```
ghcr.io/cambridge-cares/feature-info-agent:3.0.0-SNAPSHOT
```

Features:
* Added ability to set the reference time for time series queries.
* Added ability to change the limit unit for time series bounding.
* Entire class hierarchy is now queried, any point in the tree can be registered in the configuration.
* Added sample data, ontology, map, queries (etc.) to produce a testing stack.
* Added ability to inject URLs for all Blazegraph and Ontop endpoints into queries.
* Added "/refresh" route to force re-discovery of stack endpoints and re-read configurations.
* Updated configuration format to better support future changes.
* Updated documentation.
Bug Fixes:
* Updated class determination query to no longer require inheritance from OWL class.
* Major refactoring.

> [!WARNING]
> Note that just before this PR is merged, the version of the FIA will need the "-SNAPSHOT" token removed from the `pom.xml` file.

### CODE REVIEW

**Comment by cmclscourtney** [comment]
Currently the built-in stack version is 2.0.1. Need to decide if use latest or not considering this is a breaking change.

**Review by cmclscourtney — APPROVED** [review]
Tested simple example of multi-query using within the stack and worked well.

Note: Need to remove SNAPSHOT flat for fia and stack tools

**Review by sm453 — COMMENTED** [review]
Sorry about being late. Submitting this anyway.

Tested on 'real' stack. Previous functionality works now. Also tested refresh route - works.

**Comment by mdhillman** [comment]
Doesn't look like PRs can be reopened on GitHub, as such I'll take care of @sm453's late review on a new hotfix branch. Will return here and link to the new PR once it's up.

File: Agents/FeatureInfoAgent/docs/time-query-example.jpg
[the remark this answers is not in the document]
Comment (mdhillman) [reply Agents/FeatureInfoAgent/docs/time-query-example.jpg:1]: Addressed in PR #961, commit cb20b5f387a283b03276e21f555aad49ca5ff807

File: Agents/FeatureInfoAgent/examples/reporting-station.obda
[the remark this answers is not in the document]
Comment (mdhillman) [reply Agents/FeatureInfoAgent/examples/reporting-station.obda:1]: Addressed in PR #961, commit 74d417c765bbf29cbbcc363062cc340797092917

File: Agents/FeatureInfoAgent/README.md [hunk 1]
```diff
@@ -1,191 +1,218 @@
...
+The FIA is a relatively simple HTTP agent built using the TWA agent framework. Its goal is to take in the IRI of a single feature then use it to query the knowledge graphs for metadata, and the relational databases for time series data before formatting and returning it as a JSON object.
+
+At the time of writing, automatic discovery of data is not feasible, as such the developer deploying an instance of the FIA is responsible for writing SPARQL queries to both return the raw metadata as well as the data IRIs of time series data (so that these can then be looked up in the relational databases to actually get the time series data).
 
+These SPARQL queries are written on a class-by-class (T-Box) basis; this should mean that, for example, all IRIs that are A-Box instances of the `https://theworldavatar.io/ontobuildings/Building` T-Box class will reuse the same SPARQL query as they _should_ have data in the same format.
```
[the remark this answers is not in the document]
Comment (mdhillman) [reply Agents/FeatureInfoAgent/README.md:16]: Addressed in PR #961, commit 85a9b8c0e82e1cf327c034cb71bbdda5e9fe032e

**Pull request merged**